Lib/test/test_httpservers.py

```diff
@@ -1535,6 +1537,159 @@ def test_server_test_ipv4(self, _):
             server.test(ServerClass=mock_server, bind=bind)
             self.assertEqual(mock_server.address_family, socket.AF_INET)
 
+class CommandLineTestCase(unittest.TestCase):
+    def setUp(self):
+        self.default_port = 8000
+        self.default_bind = None
+        self.default_protocol = 'HTTP/1.0'
+        self.default_directory = os.getcwd()
+        self.default_handler = SimpleHTTPRequestHandler
+        self.default_server = unittest.mock.ANY
+        self.tls_cert = certdata_file('ssl_cert.pem')
+        self.tls_key = certdata_file('ssl_key.pem')
+        self.tls_password = 'somepass'
+        tls_password_file_object = tempfile.NamedTemporaryFile(mode='w+', delete=False)
+        tls_password_file_object.write(self.tls_password)
+        self.tls_password_file = tls_password_file_object.name
+        tls_password_file_object.close()
+        return super().setUp()
+
+    def tearDown(self):
+        if os.path.exists(self.tls_password_file):
+            os.remove(self.tls_password_file)
+        return super().tearDown()
+
+    def text_normalizer(self, string):
+        return textwrap.dedent(string).strip()
+
+    def invoke_httpd(self, args=[]):
+        output = StringIO()
+        with contextlib.redirect_stdout(output):
+            server._main(args)
+        return self.text_normalizer(output.getvalue())
```

Comment:
1. `text_normalizer` function is only used once, why not get rid of it
2. mutable variable as a default argument is bad practice
```suggestion
    def invoke_httpd(self, *args):
        output = StringIO()
        with contextlib.redirect_stdout(output):
            server._main(args)
        return textwrap.dedent(output.getvalue()).strip()
```